### PR TITLE
Fixing macOS build scripts and reproducibility issues

### PR DESCRIPTION
**Pull request by SeeYangZhi**
- Fixes reproducibility issue with macOS builds. 
  - Locked packages needed by pyinstaller in requirements-*.txt
- Added GA workflow for building macOS dmg.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -12,81 +12,80 @@
...
-    icons_dirname = 'pixmaps'
-    if not os.access(opts.root_path + usr_share, os.W_OK) and \
-       not os.access(opts.root_path, os.W_OK):
-        icons_dirname = 'icons'
-        if 'XDG_DATA_HOME' in os.environ.keys():
-            usr_share = os.environ['XDG_DATA_HOME']
-        else:
-            usr_share = os.path.expanduser('~/.local/share')
+if platform.system() in ["Linux", "FreeBSD", "DragonFly"]:
+    # note: we can't use absolute paths here. see #7787
```
Comment (ivan-zynesis) [inline setup.py:36]: Seems to be (one of the) root cause causing old build script failure.

**Review by ivan-zynesis — CHANGES_REQUESTED** [review]
Looks good.

Those auto formatting from your editor caused quite some unnecessary changes of `'` to `"`, shall we revert them?

**Comment by SeeYangZhi** [comment]
> 

Yeah sure, I will revert the auto-formatting.

**Review by SeeYangZhi — COMMENTED** [review]
reverted changes of quotations in latest two commits.

**Pull request merged**